libmscore/stafftype.cpp

```diff
@@ -87,6 +87,7 @@ StaffType::StaffType(StaffGroup sg, const QString& xml, const QString& name, int
       setShowBarlines(showBarLines);
       setStemless(stemless);
       setGenTimesig(genTimesig);
+      setGenKeysig(sg != StaffGroup::TAB);
```

Comment:
This doesn't resolve the issue completely: on file reading the default-constructed `StaffType` is probably created, and `read()` is called after that for it. Therefore on reading a file with a tablature or on creating a new score from template tablature staves have a staff type which has `getKeysig` flag set to `true`. Maybe setting this flag to `false` for tablature staff type could fix this.

Reply:
Done, `StaffType::write()` needs to be changed too.